src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const result = [];
+
+  function pushToResult() {
+    result.push({ ...stateClone });
+  }
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        (Object.assign(stateClone, extraData));
+        pushToResult();
+        break;
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete stateClone[key];
+        }
+        pushToResult();
+        break;
+      case 'clear':
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+        pushToResult();
+        break;
+    }
+  }
```

Comment:
You can push after switch/case once
Don't forget add default case
```suggestion

  for (const action of actions) {
    const { type, extraData, keysToRemove } = action;

    switch (type) {
      case 'addProperties':
        (Object.assign(stateClone, extraData));
        break;
      case 'removeProperties':
        for (const key of keysToRemove) {
          delete stateClone[key];
        }
        break;
      case 'clear':
        for (const key in stateClone) {
          delete stateClone[key];
        }
        break;
    }
    result.push({ ...stateClone });
  }
```